src/constraints.jl

```diff
@@ -589,26 +589,35 @@ See also [`dual`](@ref) and [`shadow_price`](@ref).
 has_duals(model::Model) = dual_status(model) != MOI.NO_SOLUTION
 
 """
-    dual(con_ref::ConstraintRef)
+    dual(con_ref::ConstraintRef; result::Int = 1)
 
 Get the dual value of this constraint in the result returned by a solver.
 Use `has_dual` to check if a result exists before asking for values.
 See also [`shadow_price`](@ref).
 """
-function dual(con_ref::ConstraintRef{Model, <:_MOICON})
-    return reshape_vector(_constraint_dual(con_ref), dual_shape(con_ref.shape))
+function dual(con_ref::ConstraintRef{Model, <:_MOICON}; result::Int = 1)
```

Comment:
I noticed performance issue with keyword argument in https://github.com/JuliaOpt/MathOptInterface.jl/pull/587
We need to check that this is not affecting performance here

Reply:
The kwarg hit on dual, if there is one, is tiny compared to everything else. This isn't going to be a bottleneck in anyone's code.